_posts/2020-12-12-cut-scotty-some-slack.md

```diff
@@ -0,0 +1,159 @@
+Do you have looming, government-imposed deadlines?
+Maybe turn your discretionary time down a notch.
+Is there a period in which lots of folks take time off and no one expects much progress?
+Go play with something new![^4]
+
+Through a lot of experimentation, I’ve developed a pretty strong internal sense of whether I’m making “enough” progress on assigned tasks, and I’ve tried to tune this sense so that it’s just a bit more pessimistic than that of anyone that I work with.
+(meaning that if *I* feel comfortable with my rate of progress, then I *probably* have more room for learning and maintenance)
+The sentiment of your teammates—as well as formal performance reviews—can be useful trailing indicators to help you train this sense.
+
+## Start building slack
+
+*“Won’t my team notice if I suddenly block out a large chunk of discretionary time?”*
+
+Yes, they will.
+And that’s why the best play here is to start small:
+spend a little bit of extra time reading an engineering blog this week.
+Go *a little bit further* out of your way with a refactor than you normally would.
+Not only are these little changes unlikely to generate much notice, they also give you the opportunity to start communicating about the *outcomes* of the time that you spend this way, which will generate trust with your teammates—allowing you to “get away” with taking slightly larger blocks of time and to start your own virtuous cycle.
+
+If you need some more specific ideas, try some of these:
+
+* Ask a more senior colleague about the books, blogs, or Twitter personalities that have had the greatest impact on their thinking.
+* Find an extensibility point in a module you’re working on that’s no longer used—and get rid of it.
+* Fix some inefficient database queries that are just slightly out of the way.
+* Investigate the history of a confusing bit of code, and either add some comments or improve its design.
+* Create a long Slack thread about something you’ve learned—or something that confuses you.
+
+Start claiming your own slack, and keep experimenting with learning and maintenance.
+Given enough time, folks might even start to think of you as a “miracle worker”.
+
+## Appendix
+
+Here’s some stuff that I considered while writing this but couldn’t quite work into the main body.
+
+TV Tropes has an entry for [“Scotty Time”](https://tvtropes.org/pmwiki/pmwiki.php/Main/ScottyTime), the phenomenon in which a quoted estimate is negated by a leader/manager, and the subordinate manages to get the job done in time.
+
+[Parkinson’s Law](https://en.wikipedia.org/wiki/Parkinson%27s_law) states that "work expands so as to fill the time available for its completion".
+A significant danger of following this strategy is that you inflate your estimates but then don’t do anything useful with the extra time.
+It’s critical to maintain discipline!
+
+“A system with no slack is a broken system” is an adage that I picked up while I worked at Faithlife;
```

Comment:
@watters, I don't suppose you'd know more about the origin of this? I feel like I might have heard it from you, first.

Reply:
It's possible. It's something I would say. It's equally possible we're both drawing from a common source.

Insufficient slack in a system (high utilization rate / high saturation) inhibits the system's ability to adapt to changes _and_ increases the amount of time wasted "waiting".

The basic idea is a common theme of much of my professional development reading of 2014–2015 which included books like:

Slack, by Tom DeMarco
The Phoenix Project, by Gene Kim
Lean Enterprise by Jez Humble, et al
High Output Management by Andy Grove

There were also countless papers and blog posts that I'd have likely read around that time on traffic congestion, work in progress/process limits, etc, particularly as they related to coordinating and prioritizing software engineering work.